utils/pattern/src/double.rs

```diff
@@ -270,6 +270,7 @@ impl PatternBackend for DoublePlaceholder {
     #[cfg(feature = "alloc")]
     type PlaceholderKeyCow<'a> = DoublePlaceholderKey;
     type Error<'a> = Infallible;
+    type StoreUtf8Error = StoreUtf8Error;
```

Comment:
Issue: it should return just Utf8Error

Reply:
There is no way to convert `PatternError` to `Utf8Error` or even to creat and instance of `Utf8Error`

Reply:
`try_store_from_utf8` should do exactly 1 thing, which is convert `[u8]` to the generic type `Store`. If you want it to also validate the pattern, then the function should be named `validate_store_utf8` or similar. But I suggested `try_store_from_utf8` since it is the minimal operation that we need.

Reply:
done